### ISSUE TITLE
Change process title if any thread is busy

### ISSUE THREAD

**Issue by nateberkopec** [puma/puma#3483]
See https://github.com/Shopify/pitchfork/blob/c514559307884e3c75f1b7abb48d4c62a0765fbd/lib/pitchfork/http_server.rb#L830-L834

not sure exactly if it's important that, e.g. 2 out of 5 threads are busy or just if any thread is busy.

**Comment by IgnacioFan** [comment]
Hi team, I am taking a look at this issue, but I'm not sure what its purpose is.

**Comment by dentarg** [comment]
So the idea comes from Pitchfork (the code @nateberkopec linked). This is the commit that added it in pitchfork https://github.com/Shopify/pitchfork/commit/3dac0a43117aa03c854b9bc30e138b60e6166929

> Include X-Request-Id header is worker proctitle
> Useful to map a process with application logs.

Pitchfork runs one process per request so that above makes sense.

Not sure how useful it is to have the Puma processes to change the title when they has busy threads. 

Seeing the discussions in https://github.com/puma/puma/pull/3517 makes me wonder if there are performance implications in doing this?

**Comment by dentarg** [comment]
My vote is on not doing this.

**Comment by dentarg** [comment]
> This is the commit that added it in pitchfork [Shopify/pitchfork@3dac0a4](https://github.com/Shopify/pitchfork/commit/3dac0a43117aa03c854b9bc30e138b60e6166929)

No, that was just an update, there is also https://github.com/Shopify/pitchfork/commit/4d87c740398ccb33d8b70e11ace21b6ca0479efb

> Set better proctitle
> This makes it easier to see the state of the pitchfork process
> tree at a glance with a simple `ps aux`.

And https://github.com/Shopify/pitchfork/commit/46f08a670a51138ee883f01fed74bf53f69d2040

> Include requests_count in workers proctitle
> Makes it easier to see at a glance how old and warmed up
> the workers are.

So Pitchfork has a lot of info in the process title :)

**Comment by casperisfine** [comment]
> So Pitchfork has a lot of info in the process title :)

Yep, something like:

```
 pitchfork (gen:0) worker[3] - requests: 42, waiting 
 pitchfork (gen:0) worker[2] - requests: 12, processing:/some/path, request_id: 78117288-46a3-4a0d-b53c-a7bcf6c6854c
```

But most of that of course doesn't make much sense in a server that respond concurrently.
